src/lib/routes/movie.ts

```diff
@@ -58,6 +59,24 @@ router.get('/search', async (req, res) => {
   res.json(movies);
 });
 
+router.get('/list', async (req, res) => {
+  const { ids } = req.query;
+  if (!ids) {
+    res.status(400).json({ message: 'Ids are required' });
+    return;
+  } else if (typeof ids !== 'string') {
+    res.status(400).json({ message: 'Ids must be a string' });
+    return;
+  }
+  const idList = ids.split(',');
+  try {
+    const movies = await getMovies(idList);
+    res.json(movies);
+  } catch (err) {
+    res.status(404).json({ error: 'Movie/s not found' });
+  }
```

Comment:
Here we catch an error but don't do anything with it, instead we send a response, which will be interpreted as a regular response by our frontend. This needs to change